sdk/identity/Azure.Identity/src/DefaultAzureCredentialFactory.cs

```diff
@@ -38,14 +38,21 @@ public TokenCredential[] CreateCredentialChain()
             {
                 if (_useDevCredentials)
                 {
-                    return
-                    [
+                    var devCredentials = new List<TokenCredential>
+                    {
                         CreateVisualStudioCredential(),
                         CreateVisualStudioCodeCredential(),
                         CreateAzureCliCredential(),
                         CreateAzurePowerShellCredential(),
                         CreateAzureDeveloperCliCredential()
-                    ];
+                    };
+
+                    if (TryCreateDevelopmentBrokerOptions(out InteractiveBrowserCredentialOptions brokerOptions))
```

Comment:
I think we want to add the credential to the chain in dev mode even if Broker is not loaded so that we can throw CredentialUnavailableException when it isn't, giving the hint for how it could be enabled.